plugins/ccgx/fu-ccgx-hpi-device.c

```diff
@@ -201,76 +182,32 @@ fu_ccgx_hpi_device_set_i2c_config (FuCcgxHpiDevice *self,
 	return TRUE;
 }
 
-static gboolean
-fu_ccgx_hpi_device_wait_for_notify (FuCcgxHpiDevice *self,
-				    guint16 *bytes_pending,
-				    GError **error)
-{
-	guint8 buf[CY_I2C_EVENT_NOTIFICATION_LEN] = { 0x0 };
-	g_autoptr(GError) error_local = NULL;
-
-	if (!g_usb_device_interrupt_transfer (fu_usb_device_get_dev (FU_USB_DEVICE (self)),
-					      self->ep_intr_in,
-					      buf, sizeof(buf), NULL,
-					      FU_CCGX_HPI_WAIT_TIMEOUT,
-					      NULL, &error_local)) {
-		g_set_error (error,
-			     FWUPD_ERROR,
-			     FWUPD_ERROR_INTERNAL,
-			     "failed to get i2c event: %s",
-			     error_local->message);
-		return FALSE;
-	}
-
-	/* @bytes_pending available on failure */
-	if (buf[0] & CY_I2C_ERROR_BIT) {
-		if (bytes_pending != NULL) {
-			if (!fu_common_read_uint16_safe (buf, sizeof(buf), 0x01,
-							 bytes_pending, G_LITTLE_ENDIAN,
-							 error))
-				return FALSE;
-		}
-		/* write */
-		if (buf[0] & 0x80) {
-			g_set_error (error,
-				     FWUPD_ERROR,
-				     FWUPD_ERROR_INTERNAL,
-				     "i2c status error in i2c write [0x%x] event: %s",
-				     (guint8) buf[0], error_local->message);
-		/* read */
-		} else {
-			g_set_error (error,
-				     FWUPD_ERROR,
-				     FWUPD_ERROR_INTERNAL,
-				     "i2c status error in i2c read [0x%x] event: %s",
-				     (guint8) buf[0], error_local->message);
-		}
-		return FALSE;
-	}
-	return TRUE;
-}
-
 static gboolean
 fu_ccgx_hpi_device_i2c_read (FuCcgxHpiDevice *self,
 			     guint8 *buf, gsize bufsz,
 			     CyI2CDataConfigBits cfg_bits,
 			     GError **error)
 {
-	guint8 i2c_status = 0x0;
+	guint8 i2c_status[CY_I2C_EVENT_NOTIFICATION_LEN] = {0};
```

Comment:
why a 3 byte buffer? fu_ccgx_hpi_device_get_i2c_status is returning just one byte...

Reply:
In fu_ccgx_hpi_device_get_i2c_status(), CY_I2C_GET_STATUS_LEN is 3 byte, so we need 3 byte buffer to read i2c status. 
guint8 i2c_status can cause stack overflow.

Reply:
Do we every use the 2nd and 3rd bytes?

Reply:
we do not use 2nd and 4rd byte in this plugin, but 1 byte read of i2c status has not been tested before.

Reply:
Not 4rd, 2nd and 3rd.